apps/microsoft-teams/frontend/src/types/appActionResults.ts

```diff
@@ -0,0 +1,74 @@
+// Types that match the backend app action result schemas
+
+export interface ListChannelsResult {
+  ok: boolean;
+  data?: ChannelResult[];
+  error?: {
+    type: string;
+    message: string;
+    details?: Record<string, unknown>;
+  };
+}
+
+export interface ChannelResult {
+  id: string;
+  tenantId: string;
+  name: string;
+  teamId: string;
+  teamName: string;
+}
+
+export interface SendTestMessageResult {
+  ok: boolean;
+  data?: {
+    messageResponseId: string;
+  };
+  error?: {
+    type: string;
+    message: string;
+    details?: Record<string, unknown>;
+  };
+}
+
+// Type guards for runtime validation
+export function isListChannelsResult(value: unknown): value is ListChannelsResult {
```

Comment:
Note: I think we can address this in a future refactor, but I'm going to (then) suggest we convert these into assertion functions rather than type predicates.